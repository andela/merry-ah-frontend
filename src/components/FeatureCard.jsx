import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FeatureButton from './FeatureButton.jsx';

/**
 * @param {function} event
 *  @returns {JSX} jsx
 */
export default class FeatureCard extends Component {
  /**
     * @param {function} render
     *  @returns {JSX} jsx
     */
  render() {
    const {
      imgUrl,
      title,
      activeClass,
    } = this.props;
    return (
      <div id="cards" className={activeClass} style={
        {
          background: `linear-gradient(rgba(0, 0, 0, 0.5), 
                        rgba(0, 0, 0, 0.5)), url(${imgUrl}) no-repeat center `,
          backgroundSize: 'cover',
        }
      }>
        <div id="feature_button_div">
          <h5>{title}</h5>
          <FeatureButton
            text="View"
            myclass={
              activeClass
                .localeCompare(
                  'cards_active',
                ) === 0 ? 'feature_button' : 'feature_button_inactive'
              } />
        </div>
      </div>
    );
  }
}
FeatureCard.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  activeClass: PropTypes.string.isRequired,
};
