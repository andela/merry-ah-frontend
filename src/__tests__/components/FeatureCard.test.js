/* eslint-disable import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import FeatureCard from '../../components/FeatureCard';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<FeatureCard
    imgUrl='../a.jpeg'
    title='title'
    activeClass='cardactive' />);
});
describe('FeatureCard Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
