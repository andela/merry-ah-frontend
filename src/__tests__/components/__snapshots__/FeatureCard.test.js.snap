// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureCard Component should match snapshot 1`] = `
<div
  className="cardactive"
  id="cards"
  style={
    Object {
      "background": "linear-gradient(rgba(0, 0, 0, 0.5), 
                        rgba(0, 0, 0, 0.5)), url(../a.jpeg) no-repeat center ",
      "backgroundSize": "cover",
    }
  }
>
  <div
    id="feature_button_div"
  >
    <h5>
      title
    </h5>
    <FeatureButton
      myclass="feature_button_inactive"
      text="View"
    />
  </div>
</div>
`;
